extensions/vertx-http/dev-console-runtime-spi/src/main/java/io/quarkus/devconsole/runtime/spi/DevConsolePostHandler.java

```diff
@@ -1,28 +1,49 @@
 package io.quarkus.devconsole.runtime.spi;
 
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
 import java.time.Duration;
 
 import io.netty.handler.codec.http.HttpHeaderNames;
 import io.netty.handler.codec.http.HttpResponseStatus;
 import io.vertx.core.Handler;
 import io.vertx.core.MultiMap;
+import io.vertx.core.http.impl.headers.VertxHttpHeaders;
 import io.vertx.ext.web.RoutingContext;
 
 public abstract class DevConsolePostHandler implements Handler<RoutingContext> {
 
     @Override
     public void handle(RoutingContext event) {
-        event.request().setExpectMultipart(true);
-        event.request().endHandler(new Handler<Void>() {
-            @Override
-            public void handle(Void ignore) {
-                dispatch(event);
+        if (event.getBody() != null) {
+            //really simple form data decoder
+            //but we can't really re-use the netty one
```

Comment:
@stuartwdouglas Just out of curiosity, why can't we reuse the `HttpPostRequestDecoder`?

Reply:
It is very much based around netty objects that we no longer have access to (i.e. it is expecting to handle a stream of io.netty.handler.codec.http.HttpContent). We could use it if we created a heap of fake Netty objects to mimic this but it did not seem worth it.

Reply:
I see. I'm fine with this approach for the dev console. We can always improve the decoder later, once needed... ;-)